### PR TITLE
Adds engiborg circuit manipulator and extra mediborg beaker holder as upgrades

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request

- Adds a new upgrade for engiborgs, a circuit apparatus that works the same as the mediborgo beaker holder, but with circuit boards. Requires iron and titanium.
- Adds an upgrade for mediborgs to get a second beaker holder. Costs metal plus glass (exactly enough for the new large beaker that comes with it).
- Moves engiborg RPED upgrade to a new node parallel with utility upgrades. The RPED shares this node with the circuit apparatus.
- Re-did the mediborg beaker holder so that most of the code is in a parent object and the beaker holder and circuit holder are children of it.

## Why It's Good For The Game

- Gives engineer borgs a new upgrade, and allows them to build computers and machines. Helpful if they'd like to repair after a breach or just work on an expansion project. Still requires a human to install, and is locked behind both tech and titanium being acquired. Does _not_ allow for picking up or using AI law boards, or other borg upgrades.

- Gives medical borgs the ability to hold two beakers, allowing them to carry two separate chems. Useful for carrying a beaker of brute-healing juice and another of burn-healing juice, or keeping a beaker of your healing chems while also carrying the cure to that virus around. Caps at the second beaker; you cannot give the borg three.

## Changelog
:cl:
add: Added a new engineer cyborg upgrade: the circuit manipulator apparatus.
add: Added a medical cyborg upgrade: the supplemental beaker storage apparatus.
balance: The engineer cyborg RPED upgrade is now on its own tech node, shared with the circuit manipulator apparatus.
/:cl:

### CODE REVIEW

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -603,3 +603,55 @@
...
+			return FALSE
+
+		E = new(R.module)
+		R.module.basic_modules += E
+		R.module.add_module(E, FALSE, TRUE)
+
+/obj/item/borg/upgrade/beaker_app/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		var/obj/item/borg/apparatus/circuit/E = locate() in R.module
```
Comment (nemvar) [inline code/game/objects/items/robot/robot_upgrades.dm]: Messed up your copy pasta

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -603,3 +603,55 @@
...
+	name = "beaker storage apparatus"
+	desc = "A supplementary beaker storage apparatus for medical cyborgs."
+	icon_state = "cyborg_upgrade3"
+	require_module = TRUE
+	module_type = /obj/item/robot_module/medical
+
+/obj/item/borg/upgrade/beaker_app/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/borg/apparatus/beaker/extra/E = locate() in R
```
Comment (nemvar) [inline code/game/objects/items/robot/robot_upgrades.dm]: You could just locate in R.module.modules and have a much smaller list to compare against.

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 1 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_upgrades.dm]: Oof

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 2 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_upgrades.dm]: The tools get moved out of the modules and into the borg itself when they're equipped. It's super bizarre, to be honest.

Comment (nemvar) [reply code/game/objects/items/robot/robot_upgrades.dm]: wtf

Comment (zxaber) [reply code/game/objects/items/robot/robot_upgrades.dm]: Oh, yeah, right, I was thinking of the contents list.

Modules would work. Actually, I just found out it'd probably work better than the current one I stole from the RPED upgrade -- as-is, it doesn't remove if the tool is equipped.

File: code/game/objects/items/robot/robot_items.dm [hunk 3]
```diff
@@ -741,3 +741,193 @@
...
+	else
+		var/image/arm = image("icon"="borg_hardware_apparatus_arm1", "layer"=FLOAT_LAYER)
+		add_overlay(arm)
+
+/obj/item/borg/apparatus/circuit/examine()
+	. = ..()
+	if(stored)
+		. += "The apparatus currently has [stored] secured."
+
+/obj/item/borg/apparatus/pre_attack(atom/A, mob/living/user, params)
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm]: shouldnt this be /obj/item/borg/apparatus/circuit

File: code/game/objects/items/robot/robot_items.dm [hunk 4]
```diff
@@ -741,3 +741,193 @@
...
+	desc = "A supplementary beaker storage apparatus."
+
+////////////////////
+//engi part holder//
+////////////////////
+
+/obj/item/borg/apparatus/circuit
+	name = "circuit manipulation apparatus"
+	desc = "A special apparatus for carrying and manipulating circuit boards. Alt-Z or right-click to drop the stored object."
+	icon = 'icons/mob/robot_items.dmi'
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm]: this is useless, the parent already has icon set for robot items

File: code/game/objects/items/robot/robot_items.dm (hunk 3 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_items.dm]: Yeah. When you stare at the same procs for hours, they all start to sorta blend together.

File: code/game/objects/items/robot/robot_items.dm [hunk 5]
```diff
@@ -741,3 +741,193 @@
...
+	. = ..()
+
+/////////////////
+//beaker holder//
+/////////////////
+
+/obj/item/borg/apparatus/beaker
+	name = "beaker storage apparatus"
+	desc = "A special apparatus for carrying beakers without spilling the contents. Alt-Z or right-click to drop the beaker."
+	icon = 'icons/mob/robot_items.dmi'
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm]: also useless

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/robot_items.dm [hunk 6]
```diff
@@ -741,3 +741,193 @@
...
+//engi part holder//
+////////////////////
+
+/obj/item/borg/apparatus/circuit
+	name = "circuit manipulation apparatus"
+	desc = "A special apparatus for carrying and manipulating circuit boards. Alt-Z or right-click to drop the stored object."
+	icon = 'icons/mob/robot_items.dmi'
+	icon_state = "borg_hardware_apparatus"
+	storable = list(/obj/item/circuitboard,
+				/obj/item/electronics/)
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm]: i dont know but i think that / is useless

**Comment by Arathianen** [comment]
Borgies are adorable and should be buffed at every opportunity.

**Comment by nemvar** [comment]
Spider borgs when?

**Comment by Arathianen** [comment]
Baste Nemvar tbqh

**Comment by optimumtact** [comment]
borg BUFF reee

File: code/game/objects/items/robot/robot_items.dm (hunk 4 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_items.dm]: fixed

File: code/game/objects/items/robot/robot_items.dm (hunk 5 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_items.dm]: fixed

File: code/game/objects/items/robot/robot_items.dm (hunk 6 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_items.dm]: fixed

File: code/game/objects/items/robot/robot_upgrades.dm (hunk 2 above)
Comment (zxaber) [reply code/game/objects/items/robot/robot_upgrades.dm]: Changed to your idea, works better than what I stole.

**Comment by Tlaltecuhtli** [comment]
>Gives engineer borgs a new upgrade, and allows them to build computers and machines.  -snip- . Still requires a human to install, -snip-

what? can they hold circuit boards or not?

**Comment by zombiegums** [comment]
> > Gives engineer borgs a new upgrade, and allows them to build computers and machines.  -snip- . Still requires a human to install, -snip-
> 
> what? can they hold circuit boards or not?

Pretty sure it can hold circuit boards. I think zxaber is saying it doesn't come pre-installed roundstart like the new mediborg beaker holder does. The circuit manipulator is an upgrade that needs to be installed into the borg by someone with hands and is gated behind research and  materials (titanium).

**Comment by MCterra10** [comment]
Speedmerge

**Comment by TetraK1** [comment]
the engiborg shackles get looser

engineers are quaking

Malf can build robo consoles for hacking and chargers that repair burn in a safe place, but both of these are true already (sat charger can be upgraded, can unwrench robo console and take it to sat/safe place).

**Comment by MCterra10** [comment]
>implying engineers ever do anything anyway

**Comment by actioninja** [comment]
Give a good reason why the limitations of borgs should be reduced even less.

**Comment by zxaber** [comment]
Allows engineer borgs to set up computers and machines after medbay gets bombed. Lets them build computers on the shuttle if the DIY shuttle was purchased. Allows them to set up new rooms and expansions on the station where computers or machines would be nice.

Often times, engiborgs are the only ones repairing damage or building the shuttle. This would be a helpful tool for those cases, while still requiring tech, mats, and a helpful human to install.

**Comment by Detective-Google** [comment]
> the engiborg shackles get looser
> 
> engineers are quaking
> 
> Malf can build robo consoles for hacking and chargers that repair burn in a safe place, but both of these are true already (sat charger can be upgraded, can unwrench robo console and take it to sat/safe place).

Being able to build robo consoles, and chargers without human intervention is pretty wack, I'd say this ought to have a limitation on what circuit boards can be used (i.e. a blacklist of most silicon related boards)

**Pull request merged**